### PR TITLE
Adds nightly image into  build pipeline

### PR DESCRIPTION
**Pull request by gurkanindibay**

### CODE REVIEW

File: .github/workflows/publish_docker_images_on_push.yml [hunk 1]
```diff
@@ -3,18 +3,17 @@ name: Build Docker Image On Push (Both Tags and Branches)
 on:
   push:
     branches:
-      - '**'
+      - "**"
     tags:
-      - '**'
+      - "**"
       -
```
Comment (hanefi) [inline .github/workflows/publish_docker_images_on_push.yml:9]:
```suggestion
```

File: .github/workflows/publish_docker_images_on_push.yml [hunk 2]
```diff
@@ -3,18 +3,17 @@ name: Build Docker Image On Push (Both Tags and Branches)
...
 jobs:
-
   build_and_publish_images:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        image_type: [latest,alpine,postgres_12 ]
+        image_type: [latest, alpine, postgres_12, nightly]
```
Comment (hanefi) [inline .github/workflows/publish_docker_images_on_push.yml:16]: IMO, this would be cleaner if separated into 4 new lines.

Comment (gurkanindibay) [reply .github/workflows/publish_docker_images_on_push.yml:15]: Fixed

File: .github/workflows/publish_docker_images_on_push.yml (hunk 1 above)
Comment (gurkanindibay) [reply .github/workflows/publish_docker_images_on_push.yml]: Fixed

File: .github/workflows/publish_docker_images_on_push.yml (hunk 2 above)
Comment (hanefi) [reply .github/workflows/publish_docker_images_on_push.yml:15]: It appears as though you forgot to push again @gurkanindibay

Comment (gurkanindibay) [reply .github/workflows/publish_docker_images_on_push.yml]: Before you suggested putting empty spaces between items. I applied that
Now I applied list structure

File: .github/workflows/publish_docker_images_cron.yml [hunk 3]
```diff
@@ -5,16 +5,12 @@ on:
...
   build_and_publish_images:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        image_type: [ latest,
-                      alpine,
-                      postgres_12,
-                      nightly ]
+        image_type: [latest, alpine, postgres_12, nightly]
```
Comment (hanefi) [inline .github/workflows/publish_docker_images_cron.yml:13]: I don't think that you understood what I meant in https://github.com/citusdata/docker/pull/256#discussion_r693115583

```suggestion
        image_type: 
        - latest
        - alpine
        - postgres_12
        - nightly
```

**Comment by hanefi** [comment]
needs a rebase

File: .github/workflows/publish_docker_images_cron.yml (hunk 3 above)
Comment (gurkanindibay) [reply .github/workflows/publish_docker_images_cron.yml]: I forgot to change here. I changed **_on_push.yml pipeline

**Review by hanefi — APPROVED** [review]

**Pull request merged**